include/envoy/network/listener.h

```diff
@@ -162,6 +163,24 @@ class ListenerConfig {
    */
   virtual const std::vector<AccessLog::InstanceSharedPtr>& accessLogs() const PURE;
 
+  /**
+   * @return the dispatcher of the master thread.
+   */
+  virtual Event::Dispatcher& dispatcher() PURE;
+
+  /**
+   * This function will be called by a worker to rebuild a filter chain placeholder.
+   */
+  virtual void
+  rebuildFilterChain(const envoy::config::listener::v3::FilterChain* const& filter_chain,
```

Comment:
To build a place holder or to build a filter chain?

Reply:
a (filter chain) placeholder will be rebuilt to a real filter chain.